nsqlookupd/http.go

```diff
@@ -265,14 +265,19 @@ func (s *httpServer) doNodes(w http.ResponseWriter, req *http.Request, ps httpro
 	producers := s.ctx.nsqlookupd.DB.FindProducers("client", "", "").FilterByActive(
 		s.ctx.nsqlookupd.opts.InactiveProducerTimeout, 0)
 	nodes := make([]*node, len(producers))
+	topicProducersMap := make(map[string]Producers)
 	for i, p := range producers {
 		topics := s.ctx.nsqlookupd.DB.LookupRegistrations(p.peerInfo.id).Filter("topic", "*", "").Keys()
 
 		// for each topic find the producer that matches this peer
 		// to add tombstone information
 		tombstones := make([]bool, len(topics))
 		for j, t := range topics {
-			topicProducers := s.ctx.nsqlookupd.DB.FindProducers("topic", t, "")
+			if _, exists := topicProducersMap[t]; !exists {
+				topicProducersMap[t] = s.ctx.nsqlookupd.DB.FindProducers("topic", t, "")
+			}
+
+			topicProducers := topicProducersMap[t]
 			for _, tp := range topicProducers {
 				if tp.peerInfo == p.peerInfo {
 					tombstones[j] = tp.IsTombstoned(s.ctx.nsqlookupd.opts.TombstoneLifetime)
```

Comment:
I think that adding a `break` just after this line is an easy way to make this slightly faster

Reply:
Done.

Reply:
PTAL.